### PR TITLE
Feature/cancel appointments

### PR DESCRIPTION
**Pull request by Aleksa1998**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 1]
```diff
@@ -434,6 +450,17 @@ public List<DoctorAppointment> FindAllValidAppointments(List<DoctorAppointment>
...
 
+        /// <summary> This method provides <c>DoctorAppointment</c> <paramref name="appointment"/> and sends it to <c>AppointmentRepository</c> there appointment.IsCanceled will be set to true. </summary>
+        /// <param name="appointment"><c>DoctorAppointment</c> is <c>DoctorAppointment</c> that needs to be canceled.
+        /// </param>
+        /// <returns>null if DoctorAppointment is not valid; otherwise, succesfully canceled DoctorAppointment. </returns>
+        public DoctorAppointment CancelAppointment(int appointmentId)
+        {
+            DoctorAppointment appointment = CheckIfAppointmentsAreInFutureFromToday(_appointmentRepository.GetByid(appointmentId));
+            if (appointment == null) return null;
+            return _appointmentRepository.CancelAppointment(appointment);
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:461]: Write return in the form of a ternary operator.

File: PatientWebApplication/PatientWebApplication/Controllers/DoctorAppointmentController.cs [hunk 2]
```diff
@@ -48,14 +48,42 @@ public IActionResult SimpleSearchAppointments(AppointmentReportSearchDto dto)
...
+        /// <summary> This method provides <paramref name="appointmentId"/> and sends it to <c>RegularAppointmentService</c> there appointment.IsCanceled will be set to true. </summary>
+        /// <param name="appointmentId"> is <c>DoctorAppointment</c> that needs to be canceled.
+        /// </param>
+        /// <returns>200 Ok with canceled appointment.</returns>
+        [HttpPut("{appointmentId}")]
+        public IActionResult CancelAppointment(int appointmentId)
+        {
+            DoctorAppointment appointment = this.regularAppointmentService.CancelAppointment(appointmentId);
+            if (appointment == null) return BadRequest();
+            return Ok(appointment);
```
Comment (trki293) [inline PatientWebApplication/PatientWebApplication/Controllers/DoctorAppointmentController.cs:84]: Write return in the form of a ternary operator.

File: PatientWebApplication/PatientWebApplication/Validators/AppointmentValidator.cs [hunk 3]
```diff
@@ -12,13 +12,13 @@ public class AppointmentValidator : AbstractValidator<DoctorAppointment>
...
         {
-            RuleFor(a => a.id).Equal(0);
+            /*RuleFor(a => a.id).Equal(0);
             RuleFor(a => a.DoctorUserId).GreaterThan(0);
             RuleFor(a => a.PatientUserId).GreaterThan(0);
             RuleFor(a => UtilityMethods.TryParseDateInCorrectFormat(a.Date)).Equal(true);
             RuleFor(a => a.Doctor).Null();
             RuleFor(a => a.Patient).Null();
-            RuleFor(a => a.referral).Empty();
+            RuleFor(a => a.referral).Empty();*/
```
Comment (trki293) [inline PatientWebApplication/PatientWebApplication/Validators/AppointmentValidator.cs:21]: Delete annotated code.

**Review by trki293 — CHANGES_REQUESTED** [review]
Correct the indicated code sections.

**Review by trki293 — APPROVED** [review]
PR is approved.

**Review by Maja0505 — APPROVED** [review]
PR is approved. Good Clean code principles were implemented.

**Pull request merged**